lib/welcome/pages/register_page.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:flutter_app/welcome/pages/login_page.dart';
+import 'package:flutter_app/util/password_field.dart';
+import 'package:flutter_app/util/supabase.dart';
+import 'package:progress_state_button/progress_button.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+class RegisterPage extends StatefulWidget {
+  const RegisterPage({super.key});
+
+  @override
+  State<RegisterPage> createState() => _RegisterPageState();
+}
+
+class _RegisterPageState extends State<RegisterPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Register"),
+      ),
+      body: const Center(
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: EdgeInsets.symmetric(horizontal: 40, vertical: 20),
+            child: RegisterForm(),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class RegisterForm extends StatefulWidget {
+  const RegisterForm({super.key});
+
+  @override
+  State<RegisterForm> createState() => _RegisterFormState();
+}
+
+class _RegisterFormState extends State<RegisterForm> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final emailController = TextEditingController();
+  final usernameController = TextEditingController();
+  final passwordController = TextEditingController();
+  final passwordConfirmationController = TextEditingController();
+  ButtonState _state = ButtonState.idle;
+
+  void onSubmit() async {
+    if (_formKey.currentState!.validate()) {
+      late final AuthResponse res;
+      try {
+        setState(() {
+          _state = ButtonState.loading;
+        });
+        res = await supabaseClient.auth.signUp(
+            password: passwordController.text, email: emailController.text);
+      } on AuthException {
+        fail();
+        showSnackBar("Something went wrong.");
+        return;
+      }
+      final User? user = res.user;
+
+      if (user != null) {
+        try {
+          await supabaseClient.from('profiles').insert({
+            'auth_id': user.id,
+            'email': user.email,
+            'username': usernameController.text,
+          });
+        } on PostgrestException {
+          fail();
+          showSnackBar("Something went wrong.");
+          return;
+        }
+        onSuccess();
+        await Future.delayed(const Duration(seconds: 2));
+        goToLogin();
+      }
+    } else {
+      fail();
+    }
+  }
+
+  void showSnackBar(String text) async {
+    ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+      content: Text(text),
+    ));
+  }
+
+  void fail() async {
+    setState(() {
+      _state = ButtonState.fail;
+    });
+    await Future.delayed(const Duration(seconds: 2));
+    setState(() {
+      _state = ButtonState.idle;
+    });
+  }
+
+  void onSuccess() async {
+    ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+      content: Text("Account created. Please log in"),
```

Comment:
can we write here something like "please confirm your mail, then you can login", so that it's clear that you have to confirm the mail before a login is possible.

Reply:
@Eddie-42 do we want to use snackbar for this at all? Maybe we should just redirect to some static page that tells you to look into your mails? Because you will get logged in via the deeplink anyways

Reply:
I don't know how other apps do this, but having a separate page seems like a hassle. It seems a little awkward for me for a user to be redirected to a page that they've never seen without any user interaction, I can imagine people getting confused how this happened and where in the app structure they are right now and, more importantly, I would be a bit annoyed by getting sent to an entirely non-functional page. I would suggest having the snackbar just read "Please log in via the e-mail link" or something and not redirecting anywhere or just to automatically open the default e-mail app?

Reply:
I think the snackbar is enough for now. We can see when we do our User study if that is unclear, then we can change it later. 
Maybe we can redirect to the welcome page?